src/components/common/CategoriesDropdown/index.tsx

```diff
@@ -1,84 +1,148 @@
-        onClick(Categories.cryptocurrency)
+      {
+        text: Categories.cryptocurrency,
+        onClick: () => {
+          onClick(Categories.cryptocurrency)
+        },
+        value: Categories.cryptocurrency,
       },
-      value: Categories.cryptocurrency,
-    },
-    {
-      text: Categories.newsAndPolitics,
-      onClick: () => {
-        onClick(Categories.newsAndPolitics)
+      {
+        text: Categories.newsAndPolitics,
+        onClick: () => {
+          onClick(Categories.newsAndPolitics)
+        },
+        value: Categories.newsAndPolitics,
       },
-      value: Categories.newsAndPolitics,
-    },
-    {
-      text: Categories.scienceAndTech,
-      onClick: () => {
-        onClick(Categories.scienceAndTech)
+      {
+        text: Categories.scienceAndTech,
+        onClick: () => {
+          onClick(Categories.scienceAndTech)
+        },
+        value: Categories.scienceAndTech,
       },
-      value: Categories.scienceAndTech,
-    },
-    {
-      text: Categories.sports,
-      onClick: () => {
-        onClick(Categories.sports)
+      {
+        text: Categories.sports,
+        onClick: () => {
+          onClick(Categories.sports)
+        },
+        value: Categories.sports,
       },
-      value: Categories.sports,
-    },
-    {
-      text: Categories.weather,
-      onClick: () => {
-        onClick(Categories.weather)
+      {
+        text: Categories.weather,
+        onClick: () => {
+          onClick(Categories.weather)
+        },
+        value: Categories.weather,
       },
-      value: Categories.weather,
-    },
-    {
-      text: Categories.miscellaneous,
-      onClick: () => {
-        onClick(Categories.miscellaneous)
+      {
+        text: Categories.miscellaneous,
+        onClick: () => {
+          onClick(Categories.miscellaneous)
+        },
+        value: Categories.miscellaneous,
       },
-      value: Categories.miscellaneous,
+    ],
+    [onClick]
+  )
+
+  const { data: categoriesFromOmen, error: categoriesError, loading: categoriesLoading } = useQuery<
+    GetCategories
+  >(queryTopCategories, {
+    context: { clientName: 'Omen' },
+    variables: {
+      first: 100,
     },
-  ]
+  })
+
+  useEffect(() => {
+    if (categoriesLoading) {
+      setCategories(Remote.loading())
+    } else if (categoriesFromOmen) {
+      const { categories } = categoriesFromOmen
+      const newCategories = [
+        ...categories.map((category) => {
+          const value = capitalizeOnlyFirstLetter(category.id)
+          return {
+            text: value,
+            onClick: () => {
+              onClick(value)
+            },
+            value: value as string,
+          }
+        }),
+        ...categoryItems,
+      ].reduce((acc: CategoryItem[], current: CategoryItem) => {
+        const x = acc.find((item) => item.text === current.text)
+        if (!x) {
+          return acc.concat([current])
+        } else {
+          return acc
+        }
```

Comment:
This is to remove repeated items, right? . We have uniqBy as a dependency to remove duplicated by key `.text`

Reply:
imported the lodash library, and  used in this block